tests/test_core.py

```diff
@@ -547,6 +547,29 @@ def test_require_variable_top_level(test_df):
     assert list(test_df['exclude']) == [False, True]
 
 
+def test_require_variable_year_list(test_df):
+    test_df.data = test_df.data[1:]
+    df = test_df
+    df.require_variable(variable='Primary Energy',
+                        year=[2005, 2010],
```

Comment:
you could have a line `years = [2005, 2010]` at the start of that function and then use this here and below, to make it more obvious to the next person that it's two different ways of testing for the same years.

And maybe add a one-line docstring like "checking for variable in any of the years" before that section and "checking for variable in all of the years" before the next one